### ISSUE TITLE
HTML static build started failing

### ISSUE THREAD

**Issue by artistsweb-denis** [gatsbyjs/gatsby#14414]
## Description

When running the `gatsby build` command it fails without a clear error explanation. Works great in development. I have a very hard time debugging this as getting to the step where static HTML gets built takes about 30 seconds in this project.

I am connecting to express server that just serves JSONs locally. Besides that the project is using a very vanilla setup of Gatsby.

### Steps to reproduce

Very hard to do so, here is a list of dependencies we are using for the project:
```
  "dependencies": {
    "axios": "0.18.0",
    "babel-plugin-styled-components": "1.10.0",
    "cors": "2.8.5",
    "express": "4.17.0",
    "gatsby": "2.3.15",
    "gatsby-background-image": "0.3.0",
    "gatsby-image": "2.0.37",
    "gatsby-plugin-manifest": "2.0.27",
    "gatsby-plugin-offline": "2.0.25",
    "gatsby-plugin-react-helmet": "3.0.12",
    "gatsby-plugin-react-svg": "2.1.1",
    "gatsby-plugin-sharp": "2.0.34",
    "gatsby-plugin-styled-components": "3.0.7",
    "gatsby-plugin-typography": "2.2.13",
    "gatsby-source-apiserver": "2.0.0",
    "gatsby-source-filesystem": "2.0.29",
    "gatsby-transformer-sharp": "2.1.18",
    "google-map-react": "1.1.4",
    "gsap": "2.1.3",
    "jquery": "3.4.1",
    "json-stringify-safe": "5.0.1",
    "normalize.css": "8.0.1",
    "prop-types": "15.7.2",
    "react": "16.8.6",
    "react-dom": "16.8.6",
    "react-helmet": "5.2.0",
    "react-input-autosize": "2.2.1",
    "react-input-range": "1.3.0",
    "react-progress-arc": "2.0.0",
    "react-slick": "0.23.2",
    "semantic-ui-css": "2.4.1",
    "semantic-ui-react": "0.86.0",
    "slick-carousel": "1.8.1",
    "typography": "0.16.19",
    "faker": "4.1.0",
    "prettier": "1.16.4",
    "styled-components": "4.2.0"
  },
```

### Expected result

Build that works.

### Actual result

```
error Building static HTML failed

See our docs page on debugging HTML builds for help https://gatsby.dev/debug-html

> 1 | import React from "react"
    | ^
  2 | import PropTypes from "prop-types"
  3 | 
  4 | export default function HTML(props) {


  WebpackError: Callback must be a function
  
  - default-html.js:1 setTimeout
    lib/.cache/default-html.js:1:1
  
  - index.js:17 Module../src/pages/index.js
    lib/src/pages/index.js:17:1
  
  - bootstrap:19 __webpack_require__
    lib/webpack/bootstrap:19:1

...... AND MORE
```

### Environment

```
System:
    OS: macOS 10.14.5
    CPU: (8) x64 Intel(R) Core(TM) i5-8259U CPU @ 2.30GHz
    Shell: 5.7.1 - /usr/local/bin/zsh
  Binaries:
    Node: 10.15.3 - /usr/local/bin/node
    Yarn: 1.15.2 - /usr/local/bin/yarn
    npm: 6.4.1 - /usr/local/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 74.0.3729.169
    Firefox: 66.0.3
    Safari: 12.1.1
  npmPackages:
    gatsby: 2.3.15 => 2.3.15 
    gatsby-background-image: 0.3.0 => 0.3.0 
    gatsby-image: 2.0.37 => 2.0.37 
    gatsby-plugin-manifest: 2.0.27 => 2.0.27 
    gatsby-plugin-offline: 2.0.25 => 2.0.25 
    gatsby-plugin-react-helmet: 3.0.12 => 3.0.12 
    gatsby-plugin-react-svg: 2.1.1 => 2.1.1 
    gatsby-plugin-sharp: 2.0.34 => 2.0.34 
    gatsby-plugin-styled-components: 3.0.7 => 3.0.7 
    gatsby-plugin-typography: 2.2.13 => 2.2.13 
    gatsby-source-apiserver: 2.0.0 => 2.0.0 
    gatsby-source-filesystem: 2.0.29 => 2.0.29 
    gatsby-transformer-sharp: 2.1.18 => 2.1.18 
```

**Comment by KyleAMathews** [comment]
What is the `HTML` function you're exporting? Is that imported or created somewhere in the component?

--
Kyle Mathews

Blog: http://bricolage.io
Twitter: http://twitter.com/kylemathews

**Comment by artistsweb-denis** [comment]
I don't use it anywhere in my code and a quick grep revealed, that it's being used from the gatsby folder in node_modules:
```
$ rg "HTML\(props" ./node_modules

./node_modules/gatsby/cache-dir/default-html.js
4:export default function HTML(props) {

./node_modules/gatsby/cache-dir/commonjs/default-html.js
12:function HTML(props) {
```

To me it seems like something is trying to use the ES6 version of a npm module, something within gatsby, as the error pointer didn't help me to locate the issue and I'm fairly certain it's not within my codebase. Do you have any idea what can cause this?

**Comment by m-allanson** [comment]
Hey @artistsweb-denis, there's a few things you can try here. 

It looks like the HTML function is coming from `default-html.js`, which is the default page chrome that's used for your site. You can override the default by creating your own copy in your `src` directory: `cp .cache/default-html.js src/html.js`. Although I'm not sure where that setTimeout is coming from.

Also try updating Gatsby and Gatsby plugins to the latest versions.

Finally are you able to create a [reproducible test case](https://www.gatsbyjs.org/contributing/how-to-make-a-reproducible-test-case/) and post it here? I know it can be time consuming to try and carve a reproduction out of a working project, but that's often the simplest way to track down mysterious problems like this.

**Comment by artistsweb-denis** [comment]
@m-allanson I updated every dependency in the project and my global instance of `gatsby-cli` all to the latest available packages in the npm registry. I am trying but I am unable to replicate the issue. We have had no problems in the previous weeks working on the project and it's a rather huge codebase now. The issue is still the same and still pointing to `lib/.cache/default-html.js:1:1`. Is there any way I can try to provide you more specific data at this point? I'm worried I will not be able to replicate the issue.

**Comment by artistsweb-denis** [comment]
@m-allanson @KyleAMathews So mysteriously I fixed it by removing just a single import from `pages/index.js` which was to **an empty JS file**. Thank you both very much for your help!

For anyone who faces these issues - it's not an easy find, took me half a day but commenting out everything and slowly adding it back page-by-page is the best solution.

**Comment by m-allanson** [comment]
How strange, I'm glad you found the issue!

> For anyone who faces these issues - it's not an easy find, took me half a day but commenting out everything and slowly adding it back page-by-page is the best solution.

👍 for this approach to isolating problems. If you're able to replicate the issue using the [`hello-world` starter](https://github.com/gatsbyjs/gatsby-starter-hello-world), then maybe we could reopen this as a bug report? Maybe we could detect and output a more helpful error message to save other people from suffering the same problem.

**Comment by artistsweb-denis** [comment]
@m-allanson I'd love to but but I failed to replicate the issue on an empty starter (although I tried for an hour and a half) and this was the most I could narrow it down to without wasting hours upon hours. Thank you very much!

**Comment by KyleAMathews** [comment]
What exactly was the index.js file? We can add a check for files like that to provide better warnings for people.

**Comment by artistsweb-denis** [comment]
@KyleAMathews sorry for the late reply. `src/pages/index.js` is our index site, which is composed of an exported functional component `IndexPage` and an exported gatsby graphql query. The import was to a JS file which had no export. It was an unnamed import - `import "motion/index.js"` which only had comments in it.

**Comment by gregwhitworth** [comment]
I'm hitting this same issue due to the [accessibility module](https://github.com/highcharts/highcharts/blob/master/js/modules/accessibility/accessibility.js). The [globals file](https://github.com/highcharts/highcharts/blob/master/js/parts/Globals.js) I think is what is triggering this issue. Here is the error specifically from Gatsby:

```
Building static HTML failed

See our docs page for more info on this error: https://gatsby.dev/debug-html


   9 |  License: www.highcharts.com/license
  10 | */
> 11 | (function(g){"object"===typeof module&&module.exports?(g["default"]=g,module.exports=g):"function"===typeof
define&&define.amd?define("highcharts/modules/accessibility",["highcharts"],function(m){g(m);g.Highcharts=m;return g}):g("undefined"!==typeof Highcharts?Highcharts:void 0)})(function(g){function m(c,g,l,k){c.hasOwnProperty(g)||(c[g]=k.apply(null,l))}g=g?g._modules:{};m(g,"modules/accessibility/KeyboardNavigationHandler.js",[g["parts/Globals.js"]],function(c){function g(c,b){this.chart=c;this.keyCodeMap=
     |
```

Any suggestions are appreciated.

**Comment by gregwhitworth** [comment]
@KyleAMathews and @m-allanson thanks to the HighCharts team I was able to get this working. I had checked their modules and they were doing the `undefined` check for their use of window and document. Even with that though, I had to add my own check as a wrapper around the method like so:

```
if (typeof window !== `undefined`) {
  highchartsAccessibility(Highcharts);
}
```
This seems a bit odd as I'm not actively using `window` in my code and the modules are all doing the correct checks. At any rate I got it working but thought I'd let you know and put it in this issue since this isn't necessarily noted in the page.
